scripts/shared/dbRequestUtils.js

```diff
@@ -192,3 +192,28 @@ gpii.dbRequest.queryDatabase = function (databaseURL, handleResponse, errorMsg,
     });
     return aRequest;
 };
+
+/**
+ * Process recursively.
+ * @param {Object} options - Object of elements required for processing. It also tracks information that need to be passed
+ * along during the processing.
+ * @param {Function} actionFunc - The action function that returns a promise whose resolved value is a number of
+ * documents to process. If the number is 0, the recursion stops. If more than 0, the recursion continues.
+ * @return {Promise} - Hold the processing result.
+ */
+gpii.dbRequest.processRecursive = function (options, actionFunc) {
```

Comment:
Now it is much better!

Reply:
Thanks to you! You are the one who wrote this function ;)